Library/Coupon.cs

```diff
@@ -71,7 +71,7 @@ public enum CouponType
         public CouponDiscountType DiscountType { get; private set; }
         public CouponState State { get; private set; }
         public RedemptionResourceType RedemptionResource { get; set; }
-        public CouponType Type { get; set; } = CouponType.SingleCode;
```

Comment:
@MuffinMan10110 what is the purpose of this change and what are it's implications?